x/ibc/05-port/types/module.go

```diff
@@ -0,0 +1,77 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/capability"
+	channelexported "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+)
+
+type IBCModule interface {
+	OnChanOpenInit(
+		ctx sdk.Context,
+		order channelexported.Order,
+		connectionHops []string,
+		portID string,
+		channelID string,
+		counterParty channeltypes.Counterparty,
+		version string,
+	) error
+
+	OnChanOpenTry(
+		ctx sdk.Context,
+		order channelexported.Order,
+		connectionHops []string,
+		portID,
+		channelID string,
+		counterparty channeltypes.Counterparty,
+		version,
+		counterpartyVersion string,
+	) error
+
+	OnChanOpenAck(
+		ctx sdk.Context,
+		portID,
+		channelID string,
+		counterpartyVersion string,
+	) error
+
+	OnChanOpenConfirm(
+		ctx sdk.Context,
+		portID,
+		channelID string,
+	) error
+
+	OnChanCloseInit(
+		ctx sdk.Context,
+		portID,
+		channelID string,
+	) error
+
+	OnChanCloseConfirm(
+		ctx sdk.Context,
+		portID,
+		channelID string,
+	) error
+
+	OnRecvPacket(
+		ctx sdk.Context,
+		packet channeltypes.Packet,
+	) (*sdk.Result, error)
+
+	OnAcknowledgementPacket(
+		ctx sdk.Context,
+		packet channeltypes.Packet,
+		acknowledment []byte,
+	) (*sdk.Result, error)
+
+	OnTimeoutPacket(
+		ctx sdk.Context,
+		packet channeltypes.Packet,
+	) (*sdk.Result, error)
+
+	GetCapability(ctx sdk.Context, name string) (capability.Capability, bool)
```

Comment:
Put Get/Claim capability callbacks as well so that ibc can request a module to provide the right capability or pass a module a capability that it can claim

Reply:
Hmm, I'm a bit confused by these - why would IBC need to request a capability from a module or vice-versa? Generally in an object-capability system it is very dangerous for a module to expose an API that allows other modules to just fetch capabilities.